pkg/tsdb/azuremonitor/metrics/url-builder.go

```diff
@@ -7,33 +7,50 @@ import (
 
 // urlBuilder builds the URL for calling the Azure Monitor API
 type urlBuilder struct {
+	ResourceURI string
+
+	// Following fields are deprecated and are not included in new queries.
+	// For backwards compat, we recreate the ResourceURI using these fields
 	DefaultSubscription string
 	Subscription        string
 	ResourceGroup       string
 	MetricDefinition    string
 	ResourceName        string
 }
 
-// Build checks the metric definition property to see which form of the url
+// BuildMetricsURL checks the metric definition property to see which form of the url
 // should be returned
-func (ub *urlBuilder) Build() string {
-	subscription := ub.Subscription
-
-	if ub.Subscription == "" {
-		subscription = ub.DefaultSubscription
-	}
-
-	metricDefinitionArray := strings.Split(ub.MetricDefinition, "/")
-	resourceNameArray := strings.Split(ub.ResourceName, "/")
-	provider := metricDefinitionArray[0]
-	metricDefinitionArray = metricDefinitionArray[1:]
-
-	urlArray := []string{subscription, "resourceGroups", ub.ResourceGroup, "providers", provider}
-
-	for i := range metricDefinitionArray {
-		urlArray = append(urlArray, metricDefinitionArray[i])
-		urlArray = append(urlArray, resourceNameArray[i])
+func (params *urlBuilder) BuildMetricsURL() string {
+	resourceURI := params.ResourceURI
+
+	// We have a legacy query from before the resource picker, so we manually create the resource URI
```

Comment:
```suggestion
	// Prior to Grafana 9, we had a legacy query object rather than a resourceURI, so we manually create the resource URI
```

Reply:
Maybe this comment is a sign that we should encapsulate this into a private method? Idk open to your thoughts though!

Reply:
here too :(

Reply:
> Maybe this comment is a sign that we should encapsulate this into a private method? Idk open to your thoughts though!

A new private method would still be visible package wide. I believe, to the extent of my Go knowledge, the only way to make it truly private would be to define the function under another package.

Also I think the legacy url building can be kept inside this function as it's fairly small.